File: BE-Movie/src/main/java/com/a2m/service/ShowtimeService.java
package com.a2m.service;

import com.a2m.entities.Movies;
import com.a2m.entities.Showtimes;
import com.a2m.model.response.ShowtimeDateResponse;
import com.a2m.model.response.ShowtimeResponse;

import java.util.Date;
import java.util.List;

public interface ShowtimeService {
    List<Showtimes> getAllShowtime();

    List<Movies> getMovieByDate(Date date);

    List<Showtimes> getShowTimeByDate(Date date, Long id);
    
//    List<Showtimes> getShowtimesByDate(String date);

    List<ShowtimeResponse> today();

    List<ShowtimeResponse> getAllShowtimeActive();

    List<ShowtimeDateResponse> findByMovie(Long idMovie);

    List<Showtimes> getShowtimeByMovieAndDate(Long idMovie, Date date);

//	List<Showtimes> getShowTimeByDate(Date date, Long id);

//	List<ShowtimeResponse> today();

}

File: FE/src/app/services/showtime/showtime.service.ts
import { HttpClient, HttpHeaders } from '@angular/common/http';
import { Injectable } from '@angular/core';
import { ApiConstant } from 'app/constants/ApiConstant';
import { PeriodicElement } from 'app/page/showtime/showtime-form/showtime-form.component';
import { HeadersUtil } from 'app/util/headers-util';
import { Observable } from 'rxjs';



@Injectable({
  providedIn: 'root'
})
export class ShowtimeService {

constructor(private http: HttpClient) { }


  today():Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/today`);
  }
  getAllShowtimes(): Observable<any> {
    return this.http.get<any>(`${ApiConstant.showtime}/getAllShowtimes`);
  }

  getMoviesByDate(date: string): Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/getMoviesByDate?date=`+date);
  }

  getAllRooms(): Observable<any> {
    return this.http.get<any>(`${ApiConstant.showtime}/getAllRooms`);
  }

  getShowTimeByDate(date: string, id: number): Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/getShowTimeByDate?date=`+date+`&id=`+id);
  }

  saveShowtime(data: any): Observable<any>{
    return this.http.post<any>(`${ApiConstant.showtime}/saveShowtimes`, data);
  }

  updateData(data: any): Observable<any>{
    return this.http.put<any>(`${ApiConstant.showtime}/updateData`, data);
  }

  getShowtimesByDate(date: string): Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/getShowtimesByDate?date=`+date);
  }

  getShowtimesByIDRoom(id: number): Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/getShowtimesByID?id=`+id);
  }

  getShowTimeByDateAndID(date: string, id: number): Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/getShowTimeByDateAndID?date=`+date+`&id=`+id);
  }
  getAllShowtimeActive():Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/all-active`);
  }

  findByMovie(idMovie: number):Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/${idMovie}`);
  }

  getShowtimeByMovieAndDate(idMovie: number, date: any):Observable<any>{
    return this.http.post<any>(`${ApiConstant.showtime}/time/${idMovie}`,date);
  }

  findById(showtimeId: any):Observable<any>{
    return this.http.get<any>(`${ApiConstant.showtime}/id/${showtimeId}`);
  }
}
